src/main/java/Problem3/Book.java

```diff
@@ -15,11 +22,19 @@ public Book(String title, String author) {
 
     // copy constructor
     public Book(Book anotherBook) {
-        // homework
+        this.title = anotherBook.title;
+        this.author = anotherBook.author;
+        this.id = anotherBook.id;
     }
 
+    // equals method for comparing variables
     @Override
-    public boolean equals(Object obj) {
-        // homework
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Book otherBook = (Book) o;
+        return id.equals(otherBook.id) &&
+                title.equals(otherBook.title) &&
```

Comment:
-5 : logic wrong. 

"A movie equals another movie only when their id are the same. Same rule for books being eqaul."